contrib/check_qps.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh -eE
+#
+# Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+
+#
+# This script detects potentially stuck Queue Pairs which have ci != pi.
+#
+
+export VFS_UCX_PATH=${VFS_UCX_PATH:=/tmp/ucx}
+export HW_CI_FILE=${HW_CI_FILE:="hw_ci"}
+export PREV_SW_PI_FILE=${PREV_SW_PI_FILE:="prev_sw_pi"}
+export QP_NUM_FILE=${QP_NUM_FILE:="qp_num"}
+# Interval between traversing QPs (in seconds)
+export QP_CHECK_INTERVAL=${QP_CHECK_INTERVAL:=10}
+# "yes" - print all QPs, "no" - print only stuck QPs
+export PRINT_ALL_QPS=${PRINT_ALL_QPS:="no"}
+
+VFS_UCX_LEVEL_2_NAME="uct"
+VFS_UCX_LEVEL_3_NAME="worker"
+VFS_UCX_LEVEL_5_NAME="iface"
+VFS_UCX_LEVEL_7_NAME="dci_pool"
+
+# Show script usage help message
+usage()
+{
+    echo " How to use "
+    echo " <variable1>=<value1> .. <variableN>=<valueN> "$0
+}
+
+if [ "$1" == "-h" ]; then
+    usage
+    exit 0
+fi
+
+declare -A qp_nums
+declare -A initial_hw_cis
+
+traverse() {
+    level=0
+
+    for file in "$1"/*
```

Comment:
maybe just something like
for file in `$1/uct/worker/*/iface/*/dci_pool/*`

Reply:
I did it by recursively traversing directories with name "uct"(level=2)/"worker"(level=3)/"iface"(level=5)/"dci_pool"(level=7) and level=1/level=4/level=6 can have any names
Then it will be easy to add support for RC, by checking "ep"(level=7) in addition to "dci_pool"
wdyt?

Reply:
imo a glob expression would be simpler than nested/recursive loops

Reply:
yes, it makes it much easier
done